### PR TITLE
Pulled in recent TF repo changes with support on collective op in eager runtime

### PR DESCRIPTION
**Pull request by mhong**
(https://github.com/tensorflow/tensorflow/commit/844dbd8730b7aba7b9f51e7455023070f8953c98).

Added a unit test with a singleton group.

Next step: make it work on a collective group of size > 1.

### CODE REVIEW

**Comment by mhong** [comment]
@swift-ci please clean test tensorflow

File: test/TensorFlowRuntime/collective.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#endif
+import StdlibUnittest
+
+var CollectiveTests = TestSuite("Collective")
+
+// TODO: Probably should be disabled for TPU execution.
+CollectiveTests.testAllBackends("SingletonGroup") {
+  let x = Tensor<Float>(1.0)
+  let handle: TensorHandle<Float> =
+    #tfop("CollectiveReduce", x,
```
Comment (pschuh) [inline test/TensorFlowRuntime/collective.swift]: There is a raw version of this op:
Raw.collectiveReduce(tensor, groupSize: 1, groupKey: 1, instanceKey: 1, mergeOp: .add, finalOp: .id, subdivOffsets: [0])

**Review by pschuh — APPROVED** [review]
LGTM. Just use the raw ops instead.

File: test/TensorFlowRuntime/collective.swift (hunk 1 above)
Comment (mhong) [reply test/TensorFlowRuntime/collective.swift]: Nice! Thanks for the suggestion. Done.

**Pull request merged**